app/src/main/java/com/CameraActivity.java

```diff
@@ -169,4 +165,104 @@ public void swapCamera() {
         mOpenCvCameraView.setCameraIndex(CameraIndex);
         mOpenCvCameraView.enableView();
     }
+
+    private Runnable runnable = new Runnable() {
+        /**
+         * This function will constantly get newest frame and proceed it with face detection
+         * and recognition.
+         */
+        @Override
+        public void run() {
+            while(true)
+            {
+                // Get frame. Set frame to null to avoid doing same operation twice.
+                Mat inputFrame = getFrame();
+                setFrame(null);
+
+                if(inputFrame != null) {
+                    MatOfRect faces = model.detectAllFaces(inputFrame);
+                    // TODO: Some way to track face, make some noise detection,
+                    // TODO: maybe some number of frames with similar object?
+
+                    // TODO: we have to proceed this input and put some number to face, and apply
+                    // TODO: face detection for frame operation.
+
+                    //Pre process all images
+                    ArrayList<Mat> faceImages = model.preProcessAllFaces(inputFrame, faces);
+
+                    // Calculate vector for each face
+                    for (Mat face : faceImages) {
+                        TensorImage image = model.changeImageRes(face);
+                        model.processImage(image);
+                    }
+
+                    // Set result synchronized.
+                    setFaces(faces);
+                }
+                else
+                {
+                    // In case of no ready frame, wait some time
+                    try {
+                        Thread.sleep(100);
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                }
+            }
+        }
+    };
+
+    /**
+     * Synchronized function to set last frame.
```

Comment:
Use Thread-safe instead of "Synchronized", as it is not that clear what you meant while reading the comment.

Reply:
+1